### PR TITLE
Working on Awkward-ready Vectors

### PR DESCRIPTION
**Pull request by henryiii**
First adaptation of work from @jpivarski

### CODE REVIEW

File: src/vector/awkward.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+class LorentzXYZ(ak.Record, LorentzXYZCommon):
+    def __repr__(self):
+        return "Lxyz({0:.3g} {1:.3g} {2:.3g} {3:.3g})".format(self.x, self.y, self.z, self.t)
+
+class LorentzXYZArray(ak.Array, LorentzXYZCommon):
+    pass
+
+
+# This function only works as a ufunc overload, but it
```
Comment (eduardo-rodrigues) [inline src/vector/awkward.py]: Seems some words got deleted

File: src/vector/core/lorentz/xyzt.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# Distributed under the 3-clause BSD license, see accompanying file LICENSE
+# or https://github.com/scikit-hep/vector for details.
+
+
+# Functions that return a value
+
+def pt(rec):
+    return np.sqrt(rec.x**2 + rec.y**2)
+
+def eta(rec):
```
Comment (eduardo-rodrigues) [inline src/vector/core/lorentz/xyzt.py]: One can be even more concise here and use `pt` in the denominator.

BTW, do we want to test for `pt==0` and return `inf`? Seems appropriate unless numpy does that for you automatically when calculating the ratii array.

**Comment by eduardo-rodrigues** [comment]
This is looking good!

File: src/vector/core/lorentz/xyzt.py (hunk 2 above)
Comment (henryiii) [reply src/vector/core/lorentz/xyzt.py]: It does:

```python
In [2]: np.array([1]) / np.array([0])
/usr/local/Cellar/ipython/7.12.0/libexec/bin/ipython:1: RuntimeWarning: divide by zero encountered in true_divide
  #!/usr/local/opt/python/bin/python3.7
Out[2]: array([inf])
```

File: src/vector/awkward.py (hunk 1 above)
Comment (henryiii) [reply src/vector/awkward.py]: That was missing in Jim's original example as well.

File: src/vector/core/lorentz/xyzt.py (hunk 2 above)
Comment (eduardo-rodrigues) [reply src/vector/core/lorentz/xyzt.py]: OK, great - had not checked.

File: src/vector/awkward.py (hunk 1 above)
Comment (eduardo-rodrigues) [reply src/vector/awkward.py]: Not a reason not to improve, right? At least let's remove the now-spurious ", but it".

**Comment by henryiii** [comment]
Putting this in as a good starting point.

**Pull request merged**